src/battle_pyramid.c

```diff
@@ -1401,27 +1401,32 @@ void GenerateBattlePyramidWildMon(void)
 
     if (reqs->nMoves != 0)
         moves = AllocZeroed(sizeof(u16) * reqs->nMoves);
-    
+
     if (reqs->nAbilities != 0)
         abilities = AllocZeroed(sizeof(u16) * reqs->nAbilities);
 
     if (round >= TOTAL_PYRAMID_ROUNDS)
         round = TOTAL_PYRAMID_ROUNDS - 1;
-    
+
     id = GetMonData(&gEnemyParty[0], MON_DATA_SPECIES, NULL) - 1;   // index in table (0-11) -> higher index is lower probability
     bstLim = 450 + (25*round) + (5*id);                             // higher BST limit for 'rarer' wild mon rolls
 
     while (1)
     {
-        species = Random() % FORMS_START;
+        species = Random() % NUM_SPECIES;
+
+        // check if base species
+        if (!GET_BASE_SPECIES_ID(species))
```

Comment:
doesn't GET_BASE_SPECIES_ID return a species ID? So shouldn't this be `if (GET_BASE_SPECIES_ID(species) != species)` ?